java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java

```diff
@@ -50,6 +50,43 @@ private void waitForJavaSourceDownloads() {
     // to filter.
     private static SearchPattern mapLocationToSearchPatternLocation(int location, String query) throws Exception {
         //TODO: #21 Normalize queries and/or verify for each location.
+
+        if (query.contains("(") || query.contains(")")) {
+            // We know that this is a list of things to loook for, broken by a | command. We should get this intra string and create an OR search pattern for each one. 
+            // ex ((FileWriter|FileReader|PrintStream|File|PrintWriter|RandomAccessFile))
+            var startListIndex = query.indexOf("(");
+            var endListIndex = query.indexOf(")");
+            logInfo("Start index: " + startListIndex + " end index: " + endListIndex);
+            var startQuery = query.substring(0, startListIndex);
+            var endQuery = query.substring(endListIndex+1, query.length());
+
+            // This sshould strip the ( ) chars
+            var listString = query.substring(startListIndex+1, endListIndex);
+            logInfo("start query: " + startQuery + " end query: " + endQuery + " list String: " + listString);
+            var list = listString.split("\\|");
+            logInfo("list: " + list);
+            ArrayList<SearchPattern> l = new ArrayList<SearchPattern>();
+            for (String s: list) {
+                var p = getPatternSingleQuery(location, startQuery + s + endQuery);
+                logInfo("Pattern Got: " + p);
+                l.add(p);
+            }
+            if (l.size() < 2) {
+                // only return the search Pattern that we got.
+                return l.get(0);
+            }
+            SearchPattern pattern = SearchPattern.createOrPattern(l.get(0), l.get(1));
+            if (l.size() > 2) {
+                for (var i = 2; i < l.size(); i++) {
+                    pattern = SearchPattern.createOrPattern(pattern, l.get(i));
+                }
+            }
+            return pattern;
+        }
```

Comment:
Another suggestion, you can use reduction for the block above, something like this:
```java
SearchPattern p = l.subList(1, l.size()).stream().reduce(l.get(0), (SearchPattern::createOrPattern));
```
It will OR all the search patterns and also care for the case when size is 1

Reply:
Man, these lambdas are really powerful. I am so used to not having these that I am writing this like go! I will try to update this PR to use this.